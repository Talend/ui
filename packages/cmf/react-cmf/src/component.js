import invariant from 'invariant';
import actionCreator from './actionCreator';
import expression from './expression';
import sagas from './sagas';
import registry from './registry';
import CONST from './constant';

/**
 * All stuff related to the routing in CMF
 * @module react-cmf/lib/component
 * @see module:react-cmf/lib/component
 */

/**
 * return a component from the registry
 * @param  {string} id the component id you want
 * @param  {object} context optional context to get the registry
 * @return {function} the react component
 */
function get(id, context) {
	const component = registry.getFromRegistry(`${CONST.REGISTRY_COMPONENT_PREFIX}:${id}`, context);
	if (!component) {
		throw new Error(`component not found in the registry: ${id}`);
	}
	return component;
}

/**
 * register a component to let CMF be able to Inject it.
 * @param  {string} id the component id you want to register
 * @param  {any} component the component you want to register
 * @param  {object} context optional context to get the registry
 */
function register(id, component, context) {
	if (!component) {
		invariant(
			process.env.NODE_ENV === 'production',
			'You cannot register undefined as a component for id "%s"',
			id,
		);
		return;
	}
	registry.addToRegistry(`${CONST.REGISTRY_COMPONENT_PREFIX}:${id}`, component, context);
	if (component.actions) {
		actionCreator.registerMany(component.actions, context);
	}
	if (component.expressions) {
		expression.registerMany(component.expressions, context);
	}
	if (component.sagas) {
		sagas.registerMany(component.sagas, context);
	}
}

const registerMany = registry.getRegisterMany(register);

function has(id, context) {
	return (
		registry.getFromRegistry(`${CONST.REGISTRY_COMPONENT_PREFIX}:${id}`, context) !== undefined
	);
}

export default {
	get,
	has,
	register,
	registerMany,
};
