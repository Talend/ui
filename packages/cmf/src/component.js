import invariant from 'invariant';
import actionCreator from './actionCreator';
import expression from './expression';
import registry from './registry';
import CONST from './constant';

/**
 * All stuff related to the routing in CMF
 * @module react-cmf/lib/component
 * @see module:react-cmf/lib/component
 */

/**
 * return a component from the registry
 * @param  {string} id the component id you want
 * @param  {object} context optional context to get the registry
 * @return {function} the react component
 */
function get(id, context) {
	const component = registry.getFromRegistry(`${CONST.REGISTRY_COMPONENT_PREFIX}:${id}`, context);
	if (!component) {
		throw new Error(`component not found in the registry: ${id}`);
	}
	return component;
}

/**
 * register a component for the router configuration
 * @param  {string} id the component id you want to register
 * @param  {any} component the component you want to register
 * @param  {object} context optional context to get the registry
 */
function register(id, component, context) {
	if (!component) {
		invariant(process.env.NODE_ENV !== 'production', 'You can register undefined as a component');
		return;
	}
	registry.addToRegistry(`${CONST.REGISTRY_COMPONENT_PREFIX}:${id}`, component, context);
	if (component.actions) {
		Object.keys(component.actions).forEach(key => {
			actionCreator.register(key, component.actions[key], context);
		});
	}
	if (component.expressions) {
		Object.keys(component.expressions).forEach(key => {
			expression.register(key, component.expressions[key], context);
		});
	}
}

const registerMany = registry.getRegisterMany(register);

function has(id, context) {
	return (
		registry.getFromRegistry(`${CONST.REGISTRY_COMPONENT_PREFIX}:${id}`, context) !== undefined
	);
}

export default {
	get,
	has,
	register,
	registerMany,
};
