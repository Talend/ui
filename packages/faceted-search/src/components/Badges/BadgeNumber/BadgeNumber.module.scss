.tc-badge-number-form {
	:global(.tc-tooltip-body) {
		margin: 0;
		padding: 2rem;
		width: 100%;
	}

	&-body {
		width: inherit;
	}

	:global(.tc-tooltip-content) {
		flex-direction: column;
	}

	:global(.tc-tooltip-footer) {
		justify-content: flex-end;
	}
}
