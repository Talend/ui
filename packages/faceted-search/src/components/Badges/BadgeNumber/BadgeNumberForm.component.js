import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { Action } from '@talend/react-components/lib/Actions';
import { getTheme } from '@talend/react-components/lib/theme';
import { Rich } from '@talend/react-components';
import { getApplyDataFeature } from '../../../helpers/usage.helpers';

import cssModule from './BadgeNumber.scss';

const theme = getTheme(cssModule);

const BadgeNumberForm = ({ id, onChange, onSubmit, value, feature, t }) => {
	const applyDataFeature = useMemo(() => getApplyDataFeature(feature), [feature]);
	const onChangeText = event => onChange(event, event.target.value);

	return (
		<form className={theme('tc-badge-number-form')} id={`${id}-number`} onSubmit={onSubmit}>
			<Rich.Layout.Body id={`${id}-badge-body`} className={theme('tc-badge-number-form-body')}>
				<input
					id={`${id}-input`}
					autoFocus // eslint-disable-line jsx-a11y/no-autofocus
					className="form-control"
					placeholder={t('TYPE_HERE', { defaultValue: 'Type here' })}
					onChange={onChangeText}
					type="number"
					value={value}
				/>
			</Rich.Layout.Body>
			<Rich.Layout.Footer id={`${id}-badge-footer`}>
				<Action
					type="submit"
					data-feature={applyDataFeature}
					label={t('APPLY', { defaultValue: 'Apply' })}
					bsStyle="info"
				/>
			</Rich.Layout.Footer>
		</form>
	);
};

BadgeNumberForm.propTypes = {
	id: PropTypes.string.isRequired,
	onChange: PropTypes.func,
	onSubmit: PropTypes.func.isRequired,
	value: PropTypes.string,
	feature: PropTypes.string.isRequired,
	t: PropTypes.func.isRequired,
};

// eslint-disable-next-line import/prefer-default-export
export { BadgeNumberForm };
