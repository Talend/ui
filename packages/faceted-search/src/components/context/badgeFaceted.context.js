import { createContext, useContext } from 'react';
import invariant from 'invariant';

const badgeFacetedContext = createContext();

const useBadgeFacetedContext = () => {
	const context = useContext(badgeFacetedContext);
	invariant(
		context,
		'[badgeFacetedContext]: You are using a badge faceted component outside its context',
	);
	return context;
};

const BadgeFacetedProvider = badgeFacetedContext.Provider;

export { BadgeFacetedProvider, useBadgeFacetedContext };
