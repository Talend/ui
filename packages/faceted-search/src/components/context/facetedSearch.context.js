import { createContext, useContext } from 'react';
import invariant from 'invariant';

const facetedSearchContext = createContext();

const useFacetedSearchContext = () => {
	const context = useContext(facetedSearchContext);
	invariant(
		context,
		'[facetedSearchContext]: You are using one of faceted mode outside the faceted search',
	);
	return context;
};

const FacetedSearchProvider = facetedSearchContext.Provider;

export { FacetedSearchProvider, useFacetedSearchContext };
