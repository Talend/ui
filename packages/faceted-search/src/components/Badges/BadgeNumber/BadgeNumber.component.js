import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import Badge from '@talend/react-components/lib/Badge';

import { BadgeFaceted } from '../BadgeFaceted';
import { BadgeNumberForm } from './BadgeNumberForm.component';
import { operatorPropTypes, operatorsPropTypes } from '../../facetedSearch.propTypes';

const createDefaultOperators = t => [
	{
		label: t('FACETED_SEARCH_DOES_NOT_CONTAINS', { defaultValue: 'Does not contains' }),
		name: 'notEquals',
		iconName: 'not-equal',
	},
	{
		label: t('FACETED_SEARCH_EQUAL', { defaultValue: 'Equal' }),
		name: 'equals',
		iconName: 'equal',
	},
	{
		label: t('FACETED_SEARCH_GREATHER_THAN', { defaultValue: 'Greater than' }),
		name: 'GreaterThan',
		iconName: 'greater-than',
	},
	{
		label: t('FACETED_SEARCH_GREATHER_THAN_OR_EQUAL', { defaultValue: 'Greater than or equal' }),
		name: 'GreaterThanOrEquals',
		iconName: 'greater-than-equal',
	},
	{
		label: t('FACETED_SEARCH_LESS_THAN', { defaultValue: 'Less than' }),
		name: 'LessThan',
		iconName: 'less-than',
	},
	{
		label: t('FACETED_SEARCH_LESS_THAN_OR_EQUAL', { defaultValue: 'Less than or equal' }),
		name: 'LessThanOrEquals',
		iconName: 'less-than-equal',
	},
];

const BadgeNumber = ({
	id,
	label,
	initialOperatorOpened,
	initialValueOpened,
	operator,
	operators,
	size,
	value,
	t,
}) => {
	const currentOperators = useMemo(() => operators || createDefaultOperators(t), [operators, t]);
	const currentOperator = operator || currentOperators[0];
	const badgeTextId = `${id}-badge-number`;
	return (
		<BadgeFaceted
			badgeId={id}
			id={badgeTextId}
			initialOperatorOpened={initialOperatorOpened}
			initialValueOpened={initialValueOpened}
			labelCategory={label}
			labelValue={value || t('FACETED_SEARCH_VALUE_ALL', { defaultValue: 'All' })}
			operator={currentOperator}
			operators={currentOperators}
			size={size}
			t={t}
			value={value || ''}
		>
			{({ onSubmitBadge, onChangeValue, badgeValue }) => (
				<BadgeNumberForm
					id={badgeTextId}
					onChange={onChangeValue}
					onSubmit={onSubmitBadge}
					value={badgeValue}
					t={t}
				/>
			)}
		</BadgeFaceted>
	);
};

BadgeNumber.propTypes = {
	label: PropTypes.string.isRequired,
	id: PropTypes.string.isRequired,
	initialOperatorOpened: PropTypes.bool,
	initialValueOpened: PropTypes.bool,
	operator: operatorPropTypes,
	operators: operatorsPropTypes,
	size: PropTypes.oneOf(Object.values(Badge.SIZES)),
	value: PropTypes.string,
	t: PropTypes.func.isRequired,
};

// eslint-disable-next-line import/prefer-default-export
export { BadgeNumber };
