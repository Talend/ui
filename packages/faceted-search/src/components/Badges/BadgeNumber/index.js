import { BadgeNumber } from './BadgeNumber.component';

export { BadgeNumber };
