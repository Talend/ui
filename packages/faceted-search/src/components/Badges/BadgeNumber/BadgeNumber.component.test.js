import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { mount } from 'enzyme';
import { BadgeNumber } from './BadgeNumber.component';
import { BadgeFacetedProvider } from '../../context/badgeFaceted.context';
import getDefaultT from '../../../translate';

const badgeFacetedContextValue = {
	onDeleteBadge: jest.fn(),
	onHideOperator: jest.fn(),
	onSubmitBadge: jest.fn(),
};

describe('BadgeNumber', () => {
	it('should mount a default badge', () => {
		// Given
		const props = {
			label: 'Price',
			id: 'myId',
			t: getDefaultT(),
		};
		// When
		const wrapper = mount(
			<BadgeFacetedProvider value={badgeFacetedContextValue}>
				<BadgeNumber {...props} />
			</BadgeFacetedProvider>,
		);
		// Then
		expect(wrapper.getElement()).toMatchSnapshot();
	});

	it('should mount a badge with some other values', () => {
		// Given
		const props = {
			id: 'customId',
			initialOpenedOperator: true,
			label: 'Price',
			value: '2981723',
			t: getDefaultT(),
		};
		// When
		const wrapper = mount(
			<BadgeFacetedProvider value={badgeFacetedContextValue}>
				<BadgeNumber {...props} />
			</BadgeFacetedProvider>,
		);
		// Then
		expect(
			wrapper
				.find('#customId-badge-number-action-overlay')
				.first()
				.text(),
		).toEqual('2981723');
	});
});
