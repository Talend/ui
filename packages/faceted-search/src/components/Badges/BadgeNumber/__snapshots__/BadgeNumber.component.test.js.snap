// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeNumber should mount a default badge 1`] = `
<BadgeNumber
  id="myId"
  label="Price"
  t={[Function]}
/>
`;
